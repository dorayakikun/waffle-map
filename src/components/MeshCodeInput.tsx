import * as React from 'react'
import {
  Dropdown,
  DropdownItemProps,
  DropdownProps,
  Input,
  InputOnChangeData,
  Message,
} from 'semantic-ui-react'

export interface Props {
  errorMessage: string
  meshCodes: string
  separator: string
  onMeshesChanged: (
    event: React.SyntheticEvent<HTMLInputElement>,
    data: InputOnChangeData
  ) => void
  onSeparatorChanged: (
    event: React.SyntheticEvent<HTMLElement>,
    data: DropdownProps
  ) => void
}

const separatorOptions = [
  {
    text: 'commas',
    value: ',',
  },
  {
    text: 'dots',
    value: '.',
  },
]

const fetchTextFrom = (options: DropdownItemProps[], value: string): string => {
  return options
    .filter(o => o.value === value)
    .map(o => o.text)
    .toString()
}

export const MeshCodeInput = (props: Props) => (
  <div>
    <Input
      error={props.errorMessage !== ''}
      fluid
      placeholder="5339-35-97.5339-35-98.5339-35-99"
      onChange={props.onMeshesChanged}
      style={{ marginTop: '10px', marginBottom: '10px' }}
      value={props.meshCodes}
    />
    {props.errorMessage !== '' && (
      <Message negative>
        <Message.Header>Waffle Map Error</Message.Header>
        <p>{props.errorMessage}</p>
      </Message>
    )}
    <Dropdown
      fluid
      onChange={props.onSeparatorChanged}
      options={separatorOptions}
      style={{ marginTop: '10px', marginBottom: '10px' }}
      text={`Split with ${fetchTextFrom(separatorOptions, props.separator)}`}
      value={props.separator}
    />
  </div>
)
