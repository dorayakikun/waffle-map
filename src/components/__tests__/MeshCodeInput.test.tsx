// @flow

import * as React from "react";
import * as render from "react-test-renderer";
import { MeshCodeInput, Props } from "../MeshCodeInput";

const defaultProps: Props = {
  errorMessage: "",
  meshCodes: "",
  onMeshesChanged: () => {},
  onSeparatorChanged: () => {},
  separator: "."
};

test("Should set props to MeshCodeInput", () => {
  const tree = render.create(<MeshCodeInput {...defaultProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Should render negative message when errorMessage is not blank", () => {
  const invalidProps = {
    ...defaultProps,
    errorMessage: "It seems there was something wrong ...",
    meshCodes: "5"
  };
  const tree = render.create(<MeshCodeInput {...invalidProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
