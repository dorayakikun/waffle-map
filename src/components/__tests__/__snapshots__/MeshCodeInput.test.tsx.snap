// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render negative message when errorMessage is not blank 1`] = `
Array [
  <div
    className="ui error fluid input"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
  >
    <input
      onChange={[Function]}
      placeholder="5339-35-97.5339-35-98.5339-35-99"
      type="text"
      value="5"
    />
  </div>,
  <div
    className="ui negative message"
  >
    <div
      className="header"
    >
      Waffle Map Error
    </div>
    <p>
      It seems there was something wrong ...
    </p>
  </div>,
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      Split with dots
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          commas
        </span>
      </div>
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          dots
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`Should set props to MeshCodeInput 1`] = `
Array [
  <div
    className="ui fluid input"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
  >
    <input
      onChange={[Function]}
      placeholder="5339-35-97.5339-35-98.5339-35-99"
      type="text"
      value=""
    />
  </div>,
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      Split with dots
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          commas
        </span>
      </div>
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          dots
        </span>
      </div>
    </div>
  </div>,
]
`;
